File: composeApp/src/commonMain/kotlin/com/checklistboteco/presentation/screen/inventory/InventoryAuditResultSheet.kt
package com.checklistboteco.presentation.screen.inventory

import androidx.compose.foundation.ExperimentalFoundationApi
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.SecondaryTabRow
import androidx.compose.material3.Tab
import androidx.compose.material3.Text
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.checklistboteco.data.remote.RemoteApplyDailyAudit
import com.checklistboteco.data.remote.RemoteInventoryAudit
import com.checklistboteco.data.remote.RemoteInventoryAuditItem
import com.checklistboteco.domain.model.InventoryAuditItemSnapshot

@OptIn(ExperimentalMaterial3Api::class, ExperimentalFoundationApi::class)
@Composable
fun InventoryAuditResultSheet(
    result: RemoteApplyDailyAudit,
    onDismiss: () -> Unit,
    modifier: Modifier = Modifier
) {
    val audit = result.audit
    val snapshots = remember(audit) {
        audit?.items.orEmpty().map { item -> item.toSnapshot(audit) }
    }
    val pageCount = 1 + snapshots.size
    val pagerState = rememberPagerState(
        initialPage = 0,
        pageCount = { pageCount }
    )
    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)

    ModalBottomSheet(
        onDismissRequest = onDismiss,
        sheetState = sheetState,
        modifier = modifier
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = 24.dp)
        ) {
            Text(
                text = if (result.alreadyApplied) "Auditoria já aplicada" else "Auditoria concluída",
                style = MaterialTheme.typography.headlineSmall,
                modifier = Modifier.padding(horizontal = 20.dp, vertical = 8.dp)
            )

            if (pageCount > 1) {
                SecondaryTabRow(
                    selectedTabIndex = pagerState.currentPage.coerceAtMost(pageCount - 1),
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Tab(
                        selected = pagerState.currentPage == 0,
                        onClick = { },
                        enabled = false,
                        text = { Text("Resumo") }
                    )
                    snapshots.forEachIndexed { index, snapshot ->
                        Tab(
                            selected = pagerState.currentPage == index + 1,
                            onClick = { },
                            enabled = false,
                            text = {
                                Text(
                                    snapshot.product.take(12),
                                    maxLines = 1
                                )
                            }
                        )
                    }
                }
            }

            HorizontalPager(
                state = pagerState,
                modifier = Modifier.fillMaxWidth()
            ) { page ->
                when (page) {
                    0 -> AuditSummaryPage(result = result, audit = audit)
                    else -> AuditItemDetailPage(snapshot = snapshots[page - 1])
                }
            }

            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 20.dp, vertical = 8.dp),
                horizontalArrangement = Arrangement.Center
            ) {
                Text(
                    text = "Página ${pagerState.currentPage + 1} de $pageCount — deslize para o lado",
                    style = MaterialTheme.typography.labelMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
        }
    }
}

@Composable
private fun AuditSummaryPage(
    result: RemoteApplyDailyAudit,
    audit: RemoteInventoryAudit?
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .verticalScroll(rememberScrollState())
            .padding(horizontal = 20.dp, vertical = 8.dp),
        verticalArrangement = Arrangement.spacedBy(8.dp)
    ) {
        if (result.alreadyApplied) {
            Text(
                "A auditoria deste dia já havia sido aplicada ao estoque administrativo.",
                style = MaterialTheme.typography.bodyMedium
            )
        }
        audit?.let {
            Text("Data: ${it.date}", style = MaterialTheme.typography.bodyMedium)
            Text("Local: ${it.location}", style = MaterialTheme.typography.bodyMedium)
            HorizontalDivider(modifier = Modifier.padding(vertical = 4.dp))
            Text("Total abertura: ${it.totalOpening}", style = MaterialTheme.typography.titleMedium)
            Text("Total vendido: ${it.totalSold}", style = MaterialTheme.typography.titleMedium)
            Text("Total saldo: ${it.totalRemaining}", style = MaterialTheme.typography.titleMedium)
            HorizontalDivider(modifier = Modifier.padding(vertical = 4.dp))
            it.items.take(15).forEach { item ->
                Text("${item.status} · ${item.product}: ${item.theoreticalRemaining} restantes")
            }
        }
        if (result.balances.isNotEmpty()) {
            HorizontalDivider(modifier = Modifier.padding(vertical = 4.dp))
            Text("Saldo administrativo", style = MaterialTheme.typography.titleMedium)
            result.balances.take(15).forEach { balance ->
                Text("${balance.productName}: ${balance.quantity}")
            }
        }
    }
}

@Composable
private fun AuditItemDetailPage(snapshot: InventoryAuditItemSnapshot) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .verticalScroll(rememberScrollState())
            .padding(horizontal = 20.dp, vertical = 8.dp),
        verticalArrangement = Arrangement.spacedBy(8.dp)
    ) {
        Text(snapshot.product, style = MaterialTheme.typography.headlineSmall)
        Text("Data: ${snapshot.auditDate}")
        Text("Local: ${snapshot.location}")
        Text("Status: ${snapshot.status}")
        HorizontalDivider(modifier = Modifier.padding(vertical = 4.dp))
        Text("Abertura: ${formatAuditNumber(snapshot.openingQuantity)}")
        Text("Vendido: ${formatAuditNumber(snapshot.soldQuantity)}")
        Text("Saldo teórico: ${formatAuditNumber(snapshot.theoreticalRemaining)}")
        HorizontalDivider(modifier = Modifier.padding(vertical = 4.dp))
        Text("Total abertura: ${formatAuditNumber(snapshot.totalOpening)}")
        Text("Total vendido: ${formatAuditNumber(snapshot.totalSold)}")
        Text("Total saldo: ${formatAuditNumber(snapshot.totalRemaining)}")
        if (snapshot.notes.isNotBlank()) {
            HorizontalDivider(modifier = Modifier.padding(vertical = 4.dp))
            Text("Observações", style = MaterialTheme.typography.titleSmall)
            Text(snapshot.notes, style = MaterialTheme.typography.bodySmall)
        }
    }
}

private fun RemoteInventoryAuditItem.toSnapshot(audit: RemoteInventoryAudit?): InventoryAuditItemSnapshot {
    return InventoryAuditItemSnapshot(
        product = product,
        auditDate = audit?.date.orEmpty(),
        location = audit?.location.orEmpty(),
        status = status,
        notes = notes,
        openingQuantity = openingQuantity,
        soldQuantity = soldQuantity,
        theoreticalRemaining = theoreticalRemaining,
        totalOpening = audit?.totalOpening ?: 0.0,
        totalSold = audit?.totalSold ?: 0.0,
        totalRemaining = audit?.totalRemaining ?: 0.0
    )
}

private fun formatAuditNumber(value: Double): String {
    return if (value % 1.0 == 0.0) value.toLong().toString() else value.toString()
}

File: Packages/Inventory/Sources/InventoryFeature/InventoryAuditResultSheet.swift
import SwiftUI
import DesignSystem

struct InventoryAuditResultSheet: View {
  let result: ApplyDailyAuditResponse
  let onDismiss: () -> Void

  @State private var selectedPage = 0

  private var audit: InventoryDailyAudit? { result.audit }
  private var snapshots: [InventoryAuditItemSnapshot] {
    guard let audit else { return [] }
    return audit.items.map { InventoryAuditItemSnapshot(item: $0, audit: audit) }
  }

  private var pageCount: Int { 1 + snapshots.count }

  var body: some View {
    NavigationStack {
      VStack(spacing: 0) {
        if pageCount > 1 {
          Picker("Página", selection: $selectedPage) {
            Text("Resumo").tag(0)
            ForEach(Array(snapshots.enumerated()), id: \.offset) { index, snapshot in
              Text(snapshot.product.prefix(12)).tag(index + 1)
            }
          }
          .pickerStyle(.segmented)
          .padding(.horizontal)
          .padding(.top, 8)
        }

        TabView(selection: $selectedPage) {
          AuditSummaryPage(result: result, audit: audit)
            .tag(0)
          ForEach(Array(snapshots.enumerated()), id: \.offset) { index, snapshot in
            AuditItemDetailPage(snapshot: snapshot)
              .tag(index + 1)
          }
        }
        .tabViewStyle(.page(indexDisplayMode: pageCount > 1 ? .automatic : .never))

        Text("Página \(selectedPage + 1) de \(pageCount) — deslize para o lado")
          .font(.caption)
          .foregroundColor(.secondary)
          .padding(.vertical, 8)
      }
      .navigationTitle(result.alreadyApplied ? "Auditoria já aplicada" : "Auditoria concluída")
      .navigationBarTitleDisplayMode(.inline)
      .toolbar {
        ToolbarItem(placement: .confirmationAction) {
          Button("Fechar", action: onDismiss)
        }
      }
    }
  }
}

private struct AuditSummaryPage: View {
  let result: ApplyDailyAuditResponse
  let audit: InventoryDailyAudit?

  var body: some View {
    List {
      if result.alreadyApplied {
        Section {
          Text("A auditoria deste dia já havia sido aplicada ao estoque administrativo.")
            .font(.footnote)
            .themedListRowBackground()
        }
      }
      if let audit {
        Section {
          LabeledContent("Data", value: audit.date)
            .themedListRowBackground()
          LabeledContent("Local", value: audit.location)
            .themedListRowBackground()
          LabeledContent("Total abertura", value: formatNumber(audit.totalOpening))
            .themedListRowBackground()
          LabeledContent("Total vendido", value: formatNumber(audit.totalSold))
            .themedListRowBackground()
          LabeledContent("Total saldo", value: formatNumber(audit.totalRemaining))
            .themedListRowBackground()
        } header: {
          themedSectionHeader("Resumo")
        }
        Section {
          ForEach(audit.items.prefix(15)) { item in
            VStack(alignment: .leading, spacing: 4) {
              Text("\(item.status) · \(item.product)")
                .font(.subheadline.weight(.semibold))
              Text("Saldo teórico: \(formatNumber(item.theoreticalRemaining))")
                .font(.caption)
                .foregroundColor(.secondary)
            }
            .themedListRowBackground()
          }
        } header: {
          themedSectionHeader("Itens")
        }
      }
      if !result.balances.isEmpty {
        Section {
          ForEach(result.balances.prefix(15)) { balance in
            LabeledContent(balance.productName, value: formatNumber(balance.quantity))
              .themedListRowBackground()
          }
        } header: {
          themedSectionHeader("Saldo administrativo")
        }
      }
    }
    .themedListStyle()
  }
}

private struct AuditItemDetailPage: View {
  let snapshot: InventoryAuditItemSnapshot

  var body: some View {
    List {
      Section {
        Text(snapshot.product)
          .font(.title2.bold())
          .themedListRowBackground()
      } header: {
        themedSectionHeader("Produto")
      }
      Section {
        LabeledContent("Data", value: snapshot.auditDate)
          .themedListRowBackground()
        LabeledContent("Local", value: snapshot.location)
          .themedListRowBackground()
        LabeledContent("Status", value: snapshot.status)
          .themedListRowBackground()
      } header: {
        themedSectionHeader("Auditoria")
      }
      Section {
        LabeledContent("Abertura", value: formatNumber(snapshot.openingQuantity))
          .themedListRowBackground()
        LabeledContent("Vendido", value: formatNumber(snapshot.soldQuantity))
          .themedListRowBackground()
        LabeledContent("Saldo teórico", value: formatNumber(snapshot.theoreticalRemaining))
          .themedListRowBackground()
      } header: {
        themedSectionHeader("Quantidades")
      }
      Section {
        LabeledContent("Total abertura", value: formatNumber(snapshot.totalOpening))
          .themedListRowBackground()
        LabeledContent("Total vendido", value: formatNumber(snapshot.totalSold))
          .themedListRowBackground()
        LabeledContent("Total saldo", value: formatNumber(snapshot.totalRemaining))
          .themedListRowBackground()
      } header: {
        themedSectionHeader("Totais do dia")
      }
      if !snapshot.notes.isEmpty {
        Section {
          Text(snapshot.notes)
            .font(.footnote)
            .themedListRowBackground()
        } header: {
          themedSectionHeader("Observações")
        }
      }
    }
    .themedListStyle()
  }
}

private func formatNumber(_ value: Double) -> String {
  if value.truncatingRemainder(dividingBy: 1) == 0 {
    return String(format: "%.0f", value)
  }
  return String(format: "%.1f", value)
}

#if os(iOS) && DEBUG
struct InventoryAuditResultSheet_Previews: PreviewProvider {
  static var previews: some View {
    InventoryAuditResultSheet(
      result: ApplyDailyAuditResponse(
        audit: InventoryDailyAudit(
          date: "2026-06-20",
          location: "Beco da Praia",
          items: [
            InventoryDailyAuditItem(
              product: "Cerveja Lata",
              status: "OK",
              notes: "Conferido.",
              openingQuantity: 120,
              soldQuantity: 45,
              theoreticalRemaining: 75
            )
          ],
          totalOpening: 500,
          totalSold: 180,
          totalRemaining: 320
        ),
        balances: [],
        alreadyApplied: false
      ),
      onDismiss: {}
    )
    .environmentObject(AppTheme.shared)
  }
}
#endif
